src/oom/oomd-util.c

```diff
@@ -385,16 +385,21 @@ int oomd_system_context_acquire(const char *proc_swaps_path, OomdSystemContext *
 int oomd_insert_cgroup_context(Hashmap *old_h, Hashmap *new_h, const char *path) {
         _cleanup_(oomd_cgroup_context_freep) OomdCGroupContext *curr_ctx = NULL;
         OomdCGroupContext *old_ctx;
+        const char *norm_path;
         int r;
 
         assert(new_h);
         assert(path);
 
-        r = oomd_cgroup_context_acquire(path, &curr_ctx);
+        norm_path = empty_to_root(path);
+
+        r = oomd_cgroup_context_acquire(norm_path, &curr_ctx);
         if (r < 0)
-                return log_debug_errno(r, "Failed to get OomdCGroupContext for %s: %m", path);
+                return log_debug_errno(r, "Failed to get OomdCGroupContext for %s: %m", norm_path);
+
+        assert(streq(norm_path, curr_ctx->path));
```

Comment:
This got reverted to the old version again.

I'll force-push with this change and @yuwata's suggestion.